wp-content/themes/core/components/section_nav/js/section-nav.js

```diff
@@ -0,0 +1,360 @@
+	if ( state.v_width >= MOBILE_BREAKPOINT && componentState.isMobile ) {
+		toggleAllSectionNavs( 'open' );
+		componentState.isMobile = false;
+		el.sectionNavs.forEach( sectionNav => handleNavFit( sectionNav ) );
+	}
+
+	// If viewport state is mobile, but component state is NOT, update component state and hide all section navs.
+	if ( state.v_width < MOBILE_BREAKPOINT && ! componentState.isMobile ) {
+		toggleAllSectionNavs( 'close' );
+		componentState.isMobile = true;
+	}
+};
+
+/**
+ * Handle Escape key events for this module.
+ *
+ * @param e
+ */
+const handleEscKeyUp = ( e ) => {
+	if ( e.key !== 'Escape' ) {
+		return;
+	}
+
+	componentState.isMobile ? toggleAllSectionNavs( 'close' ) : closeAllMoreMenus();
+};
+
+/**
+ * Handle "out" events on mobile for this module.
+ *
+ * @param sectionNav
+ * @param targetEl
+ */
+const handleOutMobile = ( sectionNav, targetEl ) => {
+	// If the section nav is not visible or the focused element is still within this section nav, bail.
+	if ( ! sectionNav.classList.contains( 'c-section-nav--visible' ) ||
+		targetEl.closest( '[data-js="c-section-nav"]' ) === sectionNav ) {
+		return;
+	}
+
+	closeSectionNav( sectionNav );
+};
+
+/**
+ * Handle "out" events on desktop for this module.
+ *
+ * @param sectionNav
+ * @param targetEl
+ */
+const handleOutDesktop = ( sectionNav, targetEl ) => {
+	const moreListItem = sectionNav.querySelector( '[data-js="c-section-nav__list-item--more"]' );
+
+	// If the More menu isn't active or the focused element is still within this more menu, bail.
+	if ( ! sectionNav.classList.contains( 'c-section-nav--more-active' ) ||
+		targetEl.closest( '[data-js="c-section-nav__list-item--more"]' ) === moreListItem ) {
+		return;
+	}
+
+	closeMoreMenu( sectionNav );
+};
+
+/**
+ * Handle click outside event for this module.
+ *
+ * @param e
+ */
+const handleClickOut = ( e ) => {
+	el.sectionNavs.forEach( ( sectionNav ) => componentState.isMobile
+		? handleOutMobile( sectionNav, e.target )
+		: handleOutDesktop( sectionNav, e.target ) );
+};
+
+/**
+ * Handle Tab outside events for this module.
+ *
+ * @param e
+ */
+const handleTabKeyUp = ( e ) => {
+	if ( e.key !== 'Tab' ) {
+		return;
+	}
+
+	el.sectionNavs.forEach( ( sectionNav ) => componentState.isMobile
+		? handleOutMobile( sectionNav, e.target )
+		: handleOutDesktop( sectionNav, e.target ) );
+};
+
+/**
+ * Build events for this module.
+ */
+const bindEvents = () => {
+	delegate( el.container, '[data-js="c-section-nav__toggle--more"]', 'click', toggleMoreMenu );
+	delegate( el.container, '[data-js="c-section-nav__toggle--more"]', 'keyup', handleEscKeyUp );
+	delegate( el.container, '[data-js="c-section-nav__toggle--mobile"]', 'click', toggleSectionNav );
+	delegate( el.container, '[data-js="c-section-nav__toggle--mobile"]', 'keyup', handleEscKeyUp );
+	delegate( el.container, '.section-nav__list-item a', 'keyup', handleEscKeyUp );
```

Comment:
❓ 
What's the diff between delegate and on here?

Reply:
the `delegate()` helper basically watches the container element and assigns the relevant event callbacks to the selector(s) any time they exist (or are added to) the document. Whereas, `on()` events will only be applied when the document is loaded. So, delegate event hooks are useful if the selector you want to add doesn't exist in the DOM when the document's load/ready methods are fired.

Reply:
Thanks!